File: prime_numbers/prime_check/prime_check.py

def is_prime(number):
	if (number <= 1):
		return False
	divider = 1
	dividers_number = 0
	limit = int(number ** 0.5) + 1
	while divider <= limit:
		if number % divider == 0:
			dividers_number += 1
		if dividers_number > 2:
			return False
		divider += 1
	return True

def main():
	number = int(input("Введите число для проверки: "))
	print("Число {}-{}".format(number, "простое" if is_prime(number) else "составное"))

if __name__ == '__main__':
	main()
	

File: prime_numbers/prime_check/prime_check.c
#include <stdio.h>
#include <math.h>

typedef unsigned long ulong;

int is_prime(ulong number){
	if (number <= 1){
		return 0;
	}
	ulong limit = (ulong)pow(number, 0.5) + 1;
	ulong divider = 1;
	unsigned int dividers_number = 0;
	while (divider <= limit){
		if (number % divider == 0){
			dividers_number++;
		}
		if (dividers_number > 2){
			return 0;
		}
		divider++;
	}
	return 1;
}

int main(){
	ulong number = 0;
	printf("Введите число для проверки: ");
	scanf("%ld", &number);
	if (is_prime(number)){
		printf("%ld-простое\n", number);
	}
	else{
		printf("%ld-составное\n", number);
	}
	return 0;
}
